corehq/apps/reports/tasks.py

```diff
@@ -368,10 +368,18 @@ def extract_form_info(form, properties=None, case_ids=case_ids):
 
     properties = set()
     if export_id:
-        schema = FormExportSchema.get(export_id)
-        for table in schema.tables:
-            # - in question id is replaced by . in excel exports
-            properties |= {c.display.replace('.', '-') for c in table.columns}
+        if export_is_legacy:
+            schema = FormExportSchema.get(export_id)
+            for table in schema.tables:
+                # - in question id is replaced by . in excel exports
+                properties |= {c.display.replace('.', '-') for c in table.columns}
+        else:
+            from corehq.apps.export.models import FormExportInstance
+            export = FormExportInstance.get(export_id)
+            for table in export.tables:
+                # - in question id is replaced by . in excel exports
+                properties |= {c.label.replace('.', '-') for c in table.columns}
```

Comment:
so close to being DRY haha (think this is fine though since it'll be just as messy trying to not copy)